### PR TITLE
Map metrics

### PR DESCRIPTION
**Pull request by haykam821**

### CODE REVIEW

File: mappings/net/minecraft/client/performance/MetricSupplier.mapping [hunk 1]
```diff
@@ -0,0 +1,11 @@
+CLASS net/minecraft/class_5948 net/minecraft/client/performance/MetricSupplier
+	FIELD field_29548 metric Lnet/minecraft/class_5951;
+	FIELD field_29549 supplier Ljava/util/function/DoubleSupplier;
+	FIELD field_29550 type Lnet/minecraft/class_5949;
+	METHOD <init> (Lnet/minecraft/class_5951;Ljava/util/function/DoubleSupplier;Lnet/minecraft/class_5949;)V
+		ARG 1 metric
+		ARG 2 supplier
+		ARG 3 type
+	METHOD method_34697 getMetric ()Lnet/minecraft/class_5951;
+	METHOD method_34698 getSupplier ()Ljava/util/function/DoubleSupplier;
```
Comment (liach) [inline mappings/net/minecraft/client/performance/MetricSupplier.mapping:10]: Hmm, `supplier.getSupplier` sounds a bit weird. What value does this supplier obtain?

Comment (haykam821) [reply mappings/net/minecraft/client/performance/MetricSupplier.mapping:10]: Double

Comment (liach) [reply mappings/net/minecraft/client/performance/MetricSupplier.mapping:10]: Hmm, this naming setup is a bit weird. Allow me to come back later, a bit busy recently.

File: mappings/net/minecraft/client/performance/MetricSupplier.mapping [hunk 2]
```diff
@@ -0,0 +1,11 @@
+CLASS net/minecraft/class_5948 net/minecraft/client/performance/MetricSupplier
```
Comment (liach) [inline mappings/net/minecraft/client/performance/MetricSupplier.mapping:1]: Imo this is better named a capturer or factory. The supplier field is only called when the metric result is generated and the supplier's return may change over time.

**Review by liach — COMMENTED** [review]
I personally recommend mapping this metrics with the whole metric system including gui support, or at least include class_5950 and 5952.

**Comment by liach** [comment]
@haykam821 would you be willing to expand this metrics pr to cover other metric components (It's a large system and imo you should map them togehter) or can someone else pick this up?

**Comment by liach** [comment]
Superseded by #2212

**Pull request closed without merging**